### PR TITLE
Fix #107 by using minLOD/maxLOD

### LINKED ISSUE #107: Layer invisible since zoom level 17
Hi

I do not know if this is a bug, but the data source (VectorTiles) is displayed right to the zoom level = 17, not above, and it should be for 0-22.  Please help.

https://codepen.io/javalonde/pen/bGqpLyM

Regards,  Mik.

### PR DESCRIPTION
**Pull request by patrickarlt**
This PR addresses #107 by preferring `minLOD` and `maxLOD` in the service where we find it. From my conversations with the JS API and vector tile teams `minLOD` and `maxLOD` define the upper and lower zooms for which tiles have been cooked. MapLibre will display zooms greater then the `maxLOD`/`maxzoom` by over-scaling the tiles.

Re: #107

### CODE REVIEW

**Comment by patrickarlt** [comment]
Note that this isn't perfect because it assumes that each of the `sources` has the same `minzoom`/`maxzoom` I did test this agianst the beta [Outdoor style](https://www.arcgis.com/home/item.html?id=659e7c1b1e374f6c8a89eefe17b23380) where the base tiles, contours and hillshade are all cooked to different levels and I didn't see any adverse rendering effects so this is likely fine for now.

File: spec/VectorBasemapLayerSpec.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-env mocha */
 // These must be vars (instead of const) due to how the unit tests are run:
```
Comment (gavinr) [inline spec/VectorBasemapLayerSpec.js:2]:
```suggestion
```

**Review by gavinr — APPROVED** [review]
Looks good - a few small cleanup items:

**Comment by gavinr** [comment]
Released in v4.0.2.

**Pull request merged**